tests/fixtures/pipeline_objects.py

```diff
@@ -466,18 +466,19 @@ def make_datastore(
 
             ############## astro_cal to create wcs ################
             if cache_dir is not None and cache_base_name is not None:
-                cache_name = cache_base_name + '.wcs.json'
-                cache_path = os.path.join(cache_dir, cache_name)
-                if os.path.isfile(cache_path):
-                    SCLogger.debug('loading WCS from cache. ')
-                    ds.wcs = WorldCoordinates.copy_from_cache(cache_dir, cache_name)
-                    prov = Provenance(
+                prov = Provenance(
                         code_version=code_version,
                         process='astro_cal',
                         upstreams=[ds.sources.provenance],
                         parameters=p.astro_cal.pars.get_critical_pars(),
                         is_testing=True,
                     )
+                cache_name = f'{cache_base_name}.wcs_{prov.id[:6]}.txt.json'
+                cache_path = os.path.join(cache_dir, cache_name)
+                if os.path.isfile(cache_path):
+                    SCLogger.debug('loading WCS from cache. ')
+                    ds.wcs = WorldCoordinates.copy_from_cache(cache_dir, cache_name)
+                    ds.wcs.load() # TODO COME BACK TO THIS, MAYBE UNNECESSARY
```

Comment:
I don't think this is needed since the "data" attribute (`wcs` in this case) would be lazy loaded when needed (probably when you call `save()`).